weather_dl/download_pipeline/fetcher.py

```diff
@@ -44,29 +44,122 @@ class Fetcher(beam.DoFn):
         client_name: The name of the download client to construct per each request.
         manifest: A manifest to keep track of the status of requests
         store: To manage where downloads are persisted.
+        optimise_download: Optimised the download.
```

Comment:
Please explain in these docs how the download is optimized.

Reply:
Related: consider using a more specific name for the flag.

Reply:
Changed it to `async_downloads`.